README.rst

```diff
@@ -1,77 +1,128 @@
-.. _curity.io/plugins: https://support.curity.io/docs/latest/developer-guide/plugins/index.html#plugin-installation
-.. _commons-codec-1.9.jar: http://central.maven.org/maven2/commons-codec/commons-codec/1.9/commons-codec-1.9.jar
-.. _commons-logging-1.2.jar: http://central.maven.org/maven2/commons-logging/commons-logging/1.2/commons-logging-1.2.jar
-.. _google-collections-1.0-rc2.jar: http://central.maven.org/maven2/com/google/collections/google-collections/1.0-rc2/google-collections-1.0-rc2.jar
-.. _httpclient-4.5.jar: http://central.maven.org/maven2/org/apache/httpcomponents/httpclient/4.5/httpclient-4.5.jar
-.. _httpcore-4.4.1.jar: http://central.maven.org/maven2/org/apache/httpcomponents/httpcore/4.4.1/httpcore-4.4.1.jar
-.. _identityserver.plugins.oauth.authenticators-utility-1.0.0.jar: https://github.com/curityio/oauth-authenticator-utility-plugin
-.. _curity.io: https://curity.io/
+* Maven 3
+* Java JDK v. 8
+
+Compiling the Plug-in from Source
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The source is very easy to compile. To do so from a shell, issue this command: ``mvn package``.
+
+Installation
+~~~~~~~~~~~~
+
+To install this plug-in, either download a binary version available from the `releases section of this project's GitHub repository <https://github.com/curityio/github-authenticator/releases>`_ or compile it from source (as described above). If you compiled the plug-in from source, the package will be placed in the ``target`` subdirectory. The resulting JAR file or the one downloaded from GitHub needs to placed in the directory ``${IDSVR_HOME}/usr/share/plugins/github``. (The name of the last directory, ``github``, which is the plug-in group, is arbitrary and can be anything.) After doing so, the plug-in will become available as soon as the node is restarted.
+
+.. note::
+
+    The JAR file needs to be deployed to each run-time node and the admin node. For simple test deployments where the admin node is a run-time node, the JAR file only needs to be copied to one location.
+
+For a more detailed explanation of installing plug-ins, refer to the `Curity developer guide <https://developer.curity.io/docs/latest/developer-guide/plugins/index.html#plugin-installation>`_.
+
+Creating an App in Github
+~~~~~~~~~~~~~~~~~~~~~~
+
+As `described in the Github documentation <https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app>`_, You can `create and register <https://github.com/settings/applications/new>`_ an OAuth App under your personal account or under any organization you have administrative access to.
+
+    .. figure:: docs/images/create-github-app1.png
+            :name: new-github-app
+            :align: center
+            :width: 500px
+
+Creating a new Github application
+
+    .. figure:: docs/images/create-github-app2.png
+            :name: create-github-app
+            :align: center
+            :width: 500px
+
+Selecting to authenticate users in the new app
```

Comment:
missing period at the end of the sentence.

Reply:
okay